### PR TITLE
stub: Correctly handle multi node file paths

### PR DESCRIPTION
**Pull request by medhefgo**

### CODE REVIEW

File: src/boot/efi/util.c [hunk 1]
```diff
@@ -769,6 +769,21 @@ EFI_STATUS make_file_device_path(EFI_HANDLE device, const char16_t *file, EFI_DE
...
+        EFI_STATUS err;
+
+        assert(dp);
+        assert(ret_str);
+
+        err = BS->LocateProtocol(&(EFI_GUID) EFI_DEVICE_PATH_TO_TEXT_PROTOCOL_GUID, NULL, (void **) &text);
+        if (err != EFI_SUCCESS)
+                return err;
+
+        *ret_str = text->ConvertDevicePathToText(dp, true, true);
```
Comment (yuwata) [inline src/boot/efi/util.c]: Please do not initialize `ret_str` on failure.

(Not a matter, but I'd like to rename `text` -> `p` or something, as `text` indicates it is a string. Also, this function has single return value. So, how about to also rename `ret_str` -> `ret`?)

File: src/boot/efi/boot.c [hunk 2]
```diff
@@ -2666,9 +2666,9 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
         if (err != EFI_SUCCESS)
                 return log_error_status_stall(err, L"Error getting a LoadedImageProtocol handle: %r", err);
 
-        loaded_image_path = DevicePathToStr(loaded_image->FilePath);
-        if (!loaded_image_path)
-                return log_oom();
+        err = device_path_to_text(loaded_image->FilePath, &loaded_image_path);
+        if (err != EFI_SUCCESS)
+                return log_error_status_stall(err, L"Error getting loaded image path: %m");
```
Comment (yuwata) [inline src/boot/efi/boot.c:2674]: "Error in getting" or "Failed to get"?

**Review by yuwata — APPROVED** [review]
LGTM. Two minor comments.

File: src/boot/efi/boot.c (hunk 2 above)
Comment (yuwata) [reply src/boot/efi/boot.c:2674]: Not addressed yet? (Feel free to ignore and push resolved button if this is correct, as I am always bad about writing English.)

Comment (medhefgo) [reply src/boot/efi/boot.c:2674]: I'm pretty sure this is proper english.

**Pull request merged**